src/network/sniffer.rs

```diff
@@ -56,55 +94,68 @@ impl Sniffer {
     }
     pub fn next(&mut self) -> Option<Segment> {
         let bytes = self.network_frames.next().ok()?;
-        let ip_packet = Ipv4Packet::new(&bytes)?;
+        // See https://github.com/libpnet/libpnet/blob/master/examples/packetdump.rs
+        let payload_offset = if self.network_interface.is_loopback() {
```

Comment:
Should this be only done on macOS if it's BPF specific?

Reply:
> Got to see IPv6 traffic with `iperf` against the loopback interface, both procfs and lsof implementations. 🎉

Did you modify the code when testing against loopback interface? I'm not sure how Linux handles loopback packets. On Mac the OS will prepend 4 bytes of header, but libpnet discards that 4 bytes and inserts 14 bytes of zero. Not sure about linux

Reply:
![image](https://user-images.githubusercontent.com/816908/72374633-f3efc880-370a-11ea-916e-ce39cc7a16f0.png)
:thinking: indeed, I did not modify the code and it worked.

But in the example linked in the comment they do it only for macos, I guess we should do the same

Reply:
Changed